pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -992,9 +1043,138 @@ func (r *ReconcileClusterPool) createCloudBuilder(pool *hivev1.ClusterPool, logg
 		cloudBuilder.Region = platform.Azure.Region
 		cloudBuilder.CloudName = platform.Azure.CloudName
 		return cloudBuilder, nil
-	// TODO: OpenStack, VMware, and Ovirt.
+	case platform.OpenStack != nil:
+		credsSecret, err := r.getCredentialsSecret(pool, platform.OpenStack.CredentialsSecretRef.Name, logger)
+		if err != nil {
+			return nil, err
+		}
+		cloudBuilder := clusterresource.NewOpenStackCloudBuilderFromSecret(credsSecret)
+		cloudBuilder.Cloud = platform.OpenStack.Cloud
+		return cloudBuilder, nil
+	// TODO: VMware, and Ovirt.
 	default:
 		logger.Info("unsupported platform")
 		return nil, errors.New("unsupported platform")
 	}
 }
+
+func (r *ReconcileClusterPool) getInventoryCustomization(pool *hivev1.ClusterPool, logger log.FieldLogger) (*corev1.LocalObjectReference, error) {
+	cdcRefs := pool.Spec.Inventory.ClusterDeploymentCustomizations
+	for _, ref := range cdcRefs {
+		cdc := &hivev1.ClusterDeploymentCustomization{}
+		refNamespacedName := types.NamespacedName{Namespace: pool.Namespace, Name: ref.Name}
+		// TODO: context should be TODO or Background?
+		// I'm not sure if this will work
+		err := r.Get(context.TODO(), refNamespacedName, cdc)
```

Comment:
I'm not sure when I should use on the other, and I see that both are being used in Hive plenty of time.

Reply:
following are the definitions

```
Background returns a non-nil, empty Context. It is never canceled, has no values, and has no deadline. It is typically used by the main function, initialization, and tests, and as the top-level Context for incoming requests.

TODO returns a non-nil, empty Context. Code should use context.TODO when it's unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a Context parameter).
```

From what I understand (I may be wrong), if there is no context passed from top-level function it is always recommended to use `context.TODO`